aws/kinesis/core/shard_map.h

```diff
@@ -74,20 +75,37 @@ class ShardMap : boost::noncopyable {
   static const std::chrono::milliseconds kMinBackoff;
   static const std::chrono::milliseconds kMaxBackoff;
 
-  void update(const std::string& start_shard_id = "");
-
-  void update_callback(
-      const Aws::Kinesis::Model::DescribeStreamOutcome& outcome);
+  void update();
+  void list_shards(const std::string& next_token = "");
+  void list_shards_callback(const Aws::Kinesis::Model::ListShardsOutcome& outcome);
 
   void update_fail(const std::string& code, const std::string& msg = "");
 
+  void clear_all_stored_shards() {
+    end_hash_key_to_shard_id_.clear();
+    open_shard_ids_.clear();
+  }
+
+  void store_open_shard(const uint64_t shard_id, const uint128_t end_hash_key) {
```

Comment:
Consider moving these to .cc as they will be inlined and they are only used in the cc file.

Reply:
Will do.